strawberryfields/backends/fockbackend/tests/test_state_preparations.py

```diff
@@ -0,0 +1,161 @@
+
+import pytest
+
+import numpy as np
+
+
+MAG_ALPHAS = np.linspace(0, .8, 4)
+PHASE_ALPHAS = np.linspace(0, 2 * np.pi, 7, endpoint=False)
+NBARS = np.linspace(0, 5)
+BATCH_SIZE = 5
+SEED = 143
+
+class TestRepresentationIndependent:
+    """Basic implementation-independent tests."""
+
+    def test_prepare_vac(self, setup_backend, tol):
+        """Tests the ability to prepare vacuum states."""
+        backend = setup_backend(1)
+
+        backend.prepare_vacuum_state(0)
+        assert np.all(backend.is_vacuum(tol))
+
+    @pytest.mark.parametrize("mag_alpha", MAG_ALPHAS)
+    @pytest.mark.parametrize("phase_alpha", PHASE_ALPHAS)
+    def test_fidelity_coherent(self, setup_backend, mag_alpha, phase_alpha, tol):
+        """Tests if a range of coherent states have the correct fidelity."""
+
+        backend = setup_backend(1)
+
+        alpha = mag_alpha * np.exp(1j * phase_alpha)
+        backend.prepare_coherent_state(alpha, 0)
+        state = backend.state()
+        assert np.allclose(state.fidelity_coherent([alpha]), 1., atol=tol, rtol=0.)
+
+    @pytest.mark.parametrize("nbar", NBARS)
+    def test_prepare_thermal_state(self, setup_backend, nbar, cutoff, tol):
+        """Tests if thermal states with different nbar values
+        give the correct fock probabilities"""
+
+        backend = setup_backend(1)
+
+        backend.prepare_thermal_state(nbar, 0)
+        ref_probs = np.array([nbar ** n / (nbar + 1) ** (n + 1) for n in range(cutoff)])
+        state = backend.state()
+        state_probs = np.array([state.fock_prob([n]) for n in range(cutoff)]).T # transpose needed for array broadcasting to work in batch mode
+        assert np.allclose(ref_probs, state_probs, atol=tol, rtol=0.)
+
+
+class TestFockRepresentation:
+    """Tests that make use of the Fock basis representation."""
+
+    def test_normalized_prepare_vac(self, setup_backend, tol):
+        """Tests the ability to prepare vacuum states."""
+
+        backend = setup_backend(1)
+
+        backend.prepare_vacuum_state(0)
+        state = backend.state()
+        tr = state.trace()
+        assert np.allclose(tr, 1., atol=tol, rtol=0.)
+
+    @pytest.mark.parametrize("mag_alpha", MAG_ALPHAS)
+    @pytest.mark.parametrize("phase_alpha", PHASE_ALPHAS)
+    def test_normalized_coherent_state(self, setup_backend, mag_alpha, phase_alpha, tol):
+        """Tests if a range of coherent states are normalized."""
+
+        alpha = mag_alpha * np.exp(1j * phase_alpha)
+        if alpha == 0.:
+            pass
+        else:
+            backend = setup_backend(1)
+
+            backend.prepare_coherent_state(alpha, 0)
+            state = backend.state()
+            tr = state.trace()
+            assert np.allclose(tr, 1., atol=tol, rtol=0.)
+
+
+    def test_prepare_ket_state(self, setup_backend, cutoff, tol):
+        """Tests if a ket state with arbitrary parameters is correctly prepared."""
+        np.random.seed(SEED)
+        for _ in range(10):
```

Comment:
Why is this test repeated 10 times? As far as I can tell, should only need to be run once.

Reply:
I think whoever wrote the the test was looking to increase the coverage of possible state preparations by randomly choosing a bunch. I'll refactor to make it run once and also not use any random steps

Reply:
Actually, let's mark removing randomness as a different issue (to be resolved globally wherever possible)

Reply:
I will just remove the iteration here

Reply:
Sounds good :+1: